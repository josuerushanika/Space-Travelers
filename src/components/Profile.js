import { useState, useEffect } from 'react';
import { useSelector } from 'react-redux';
import '../style/Profile.css';

const Profile = () => {
  const { missions } = useSelector((state) => state.missions);
  const rockets = useSelector((state) => state.rockets.rocketListData);
  const [joinedMissions, setJoinedMissions] = useState([]);
  const [reservedRockets, setReservedRockets] = useState([]);

  useEffect(() => {
    const filteredMissions = missions.filter((mission) => mission.joined);
    setJoinedMissions(filteredMissions);
  }, [missions]);

  useEffect(() => {
    const filteredRockets = rockets.filter((rocket) => rocket.reserved);
    setReservedRockets(filteredRockets);
  }, [rockets]);

  return (
    <div className="profile-main">
      <div className="missionBox">
        <h1 className="selected-header">My Missions</h1>
        {joinedMissions.length ? (
          <div className="joinedMission">
            {joinedMissions.map((mission) => (
              <div key={mission.mission_id} className="joinedMissions">
                {mission.mission_name}
              </div>
            ))}
          </div>
        ) : (
          <div className="noJoined">No Missions Joined</div>
        )}
      </div>

      <div className="missionBox">
        <h1 className="selected-header">My Rockets</h1>
        {reservedRockets.length ? (
          <div className="reservedRockets">
            {reservedRockets.map((rocket) => (
              <div key={rocket.rocket_id} className="joinedMissions">
                {rocket.name}
              </div>
            ))}
          </div>
        ) : (
          <div className="noReserved">No Rockets Reserved</div>
        )}
      </div>
    </div>
  );
};

export default Profile;
