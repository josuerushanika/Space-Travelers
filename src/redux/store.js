import { configureStore } from '@reduxjs/toolkit';
import missionReducer from './mission/missionSlice';
import rocketReducer from './rockets/rocketSlice';

const store = configureStore({
  reducer: {
    rockets: rocketReducer,
    missions: missionReducer,
  },
});

export default store;

