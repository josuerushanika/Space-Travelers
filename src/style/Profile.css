.profile-main {
  display: flex;
  justify-content: space-between;
  padding: 0% 2%;
}

.missionBox {
  /* border: red solid; */
  width: 48%;
}

.selected-header {
  margin-bottom: 3%;
}

.joinedMissions {
  border: solid 1px rgb(151, 151, 151);
  padding: 4%;
  font-size: 1.5rem;
}
