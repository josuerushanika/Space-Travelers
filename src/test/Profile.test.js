import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import Profile from '../components/Profile';
import store from '../redux/store';

describe('Profile', () => {
  it('renders Profile', () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <BrowserRouter>
            <Profile />
          </BrowserRouter>
        </Provider>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
