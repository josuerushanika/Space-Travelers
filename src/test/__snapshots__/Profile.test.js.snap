// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Profile renders Profile 1`] = `
<div>
  <div
    className="missionBox"
  >
    <h1>
      My Missions
    </h1>
    <div
      className="noJoined"
    >
      No Missions Joined
    </div>
  </div>
  <div
    className="missionBox"
  >
    <h1>
      My Rockets
    </h1>
    <div
      className="noReserved"
    >
      No Rockets Reserved
    </div>
  </div>
</div>
`;
